common/include/joy.h

```diff
@@ -50,19 +50,28 @@ namespace dcx {
 #if DXX_MAX_BUTTONS_PER_JOYSTICK
 extern window_event_result joy_button_handler(SDL_JoyButtonEvent *jbe);
 #else
-#define joy_button_handler(jbe)	static_cast<void>(static_cast<SDL_JoyButtonEvent *const &>(jbe))
+static inline window_event_result joy_button_handler(void *)
```

Comment:
This should not be `void *`.  I suggest preserving the old macro definition, but with a tweak to produce the return value you need:
```c++
#define joy_button_handler(jbe)	(static_cast<SDL_JoyButtonEvent *const &>(jbe), window_event_result::ignored)
```